src/core/src/main/java/org/apache/jmeter/gui/action/template/Template.java

```diff
@@ -150,47 +157,37 @@ public boolean equals(Object obj) {
     }
 
     private boolean mapsEquals(Map<String, String> map1, Map<String, String> map2) {
```

Comment:
Is this method required at all?
Its implementation is exactly the same as `Map#equals`. So could we just use `Objects.equal(map1, map2)`?

Reply:
Good question. I only formatted it, happy to revert the changes - wouldn't want to make any changes as part of this PR unless the test coverage was really good.

Reply:
I've refactored and added more tests :)